src/Uploader/useUploader.ts

```diff
@@ -77,11 +77,31 @@ export default function useUploader({
     const bytesToMb = maxFileSize / 1024 / 1024;
     const limitSize = `${bytesToMb.toFixed()} MB`;
 
+    // check if the uploader is for images
+    const isImageUploader = Object.keys(acceptedFiles ?? {}).some(key =>
+      key.includes("image/")
+    );
+
     // set error message based on error code
-    const errorMessage =
-      errorCode === "file-invalid-type"
-        ? "File type must be a .gif, .jpg, .jpeg, .png, .webp"
-        : `File size exceeds the limit of ${limitSize}`;
+    let errorMessage = "";
+    switch (errorCode) {
+      case "file-invalid-type":
+        errorMessage = isImageUploader
+          ? "File type must be a .gif, .jpg, .jpeg, .png, .webp"
+          : fileRejection[0].errors[0].message;
+        break;
+      case "file-too-large":
+        errorMessage = `File size exceeds the limit of ${limitSize}.`;
+        break;
+      case "too-many-files":
+        errorMessage = `You can only upload ${filesLimit} file(s).`;
+        break;
+      case "file-too-small":
+        errorMessage = `File is too small. Please upload a larger file.`;
```

Comment:
Curious, how is this error ever thrown? How is a small file defined? Do we even have props for that in the component?

Reply:
Yes it is very unlikely as we dont have a prop for this. Had a chat with @Sowbhagya-ipg abou this, we have added this error message to align with others, in future if we add a prop we can update this to `File is too small. Please upload a file larger than {x}Mb`. May be it needed to mention as a commnent for future reference.

Reply:
Presumably we should also be showing the sise limit in the error message like we do for the file-too-large. I don't like the fact you've got code here tha tcan never be hit, it's effectively dead code. We also can't test it presumably. I suggest you remove it.